File: tada/src/main/java/com/qadri/tada/entity/TaskEntity.java
package com.qadri.tada.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class TaskEntity {
    @Id
    private Long id;

    @Column(nullable = false)
    private String taskName;

    private String taskDescription;

    @Column(nullable = false)
    private Boolean isCompleted;

    @Column(nullable = false)
    private Boolean isDeleted;

    @UpdateTimestamp
    private Instant lastUpdateTime;

    @CreationTimestamp
    @Column(updatable = false)
    private Instant createdAt;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: android-app/app/src/main/java/com/qadri/to_do/data/room/TaskEntity.kt
package com.qadri.to_do.data.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity(tableName = "task")
data class TaskEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long,
    @ColumnInfo(name = "task_name")
    val taskName: String,
    @ColumnInfo(name = "task_description")
    val taskDescription: String,
    @ColumnInfo(name = "is_completed")
    val isCompleted: Boolean,
    val isSynced: Boolean = false,
    val isDeleted: Boolean = false,
    val lastUpdateTime: Long,
    val createdAt: Long
)
